### PR TITLE
MX: add bank holidays

### PR DESCRIPTION
**Pull request by gerryster**
This completes the set of bank holidays according to the Mexican government and the Bank of Mexico. See sources below.

Note that I do have a decent knowledge of Spanish and was able to understand the legislation cited below. I've also provided an English document for reference.

### CODE REVIEW

File: data/countries/MX.yaml [hunk 1]
```diff
@@ -1,5 +1,19 @@
 holidays:
   # @attrib https://en.wikipedia.org/wiki/Public_holidays_in_Mexico
+  #
+  # Bank Holidays:
```
Comment (gerryster) [inline data/countries/MX.yaml]: This is a bit verbose but my goal here is to help myself (or someone else) in the future to be able to find the 2020 dates when they are published. The pattern of which dates are banking holidays has not changed in the past three years.

Comment (commenthol) [reply data/countries/MX.yaml]: Great. Then let's use the pattern until it may change...

File: data/countries/MX.yaml [hunk 2]
```diff
@@ -64,6 +64,9 @@ holidays:
         type: observance
       09-16:
         _name: Independence Day
+      11-2:
```
Comment (commenthol) [inline data/countries/MX.yaml]: Can you please change to `11-02:` ?

**Comment by commenthol** [comment]
Look good to me. Could you please change the "All Souls' Day" to `11-02` with a leading `0`? The parser seams to process it, but for conformance reasons I would prefer the leading zero.

File: data/countries/MX.yaml (hunk 2 above)
Comment (gerryster) [reply data/countries/MX.yaml]: Oh, good call. I've updated this.

**Comment by commenthol** [comment]
Thanks...

**Pull request merged**